### ISSUE TITLE
Общая концепция

### ISSUE THREAD

**Issue by hohlick** [hohlick/PowerBOOK#3]
Несколько вопросов по общей концепции книги:

1. **Идея**: Это должен быть полный гайд по PQ от азов до ... , или учебник по написанию кода?
2. **Уровень книги**: на кого рассчитываем? Более-менее продвинутый пользователь, знакомый с интерфейсом? Или новичок? Не очень хочется тратить много времени на самый начальный уровень, хотя, может быть, и надо. Вариант: начинаем с азов и развиваем до +/- продвинутого уровня (Data Connectors в VS, например)
3. **Акцент**: на чем? Или без акцента?

**Comment by maxim-uvarov** [comment]
Мне сейчас нравится идея, чтобы не писать книгу целиком, а дописывать по мере появления новой информации. Чтобы сразу шарить можно было. 

Мне кажется проще всего с FAQ начать.

**Comment by hohlick** [comment]
Я согласен, что целиком нет смысла ее кропать и потом выкладывать. Надо постить части по готовности, и даже по неготовности, чтобы было о чем думать и обсуждать/править. Но, скажем, так, тут два момента:

**Структура книги** - то есть, её хребет. Его стоит продумать заранее. Как, например, делал Reza Rad в своей онлайн-книге. Он набросал структуру, затем по возможности дополнял ее, в том числе постами своими. Мне кажется, для опен-сорсного проекта это самое то - если вдруг найдутся желающие, смогут поучаствовать.

Ну и общий **концепт**. Наверное, в каждом крупном разделе должны быть части теоретическая, разные _How-to_ плюс ссылки на основные материалы по теме. Ну или что-то похожее. 

То есть и не википедия, и не академия.

**Comment by hohlick** [comment]
> с FAQ начать.
С FAQ по PQ? Можно и с него.
В любом случае структура - гибкая. Можем в любой момент переделать и передумать.

**Comment by maxim-uvarov** [comment]
Макс, если у тебя есть мысли по структуре - делись:) если нет, то можно в хаотичном порядке заполнять главы: 
- power query, 
- Dax, 
- визуальное представление данных, 
- power bi service
- data gateway

**Comment by maxim-uvarov** [comment]
Но с самого начала нужно ответить на вопрос: какая цель у книги?

Я это понимаю как «помощь в освоении инструмента», бесплатно и качественно. Макс, ты как?

**Comment by hohlick** [comment]
Вот, важный вопрос, который стоило обсудить в начале :) Цель книги, и вообще - о чём она?

1. Ты предлагаешь делать книгу по Power BI, а я изначально думал только о Power Query.
2. Помощь в освоении - то есть базовые концепты? или глубокое погружение?

В итоге получается, что если Power BI в целом - то это должна быть либо реально а-ля вики, референс-лист и базовые вещи, либо, если погружаемся в детали, мы тут и погибнем под объемом. Впрочем, не так страшен черт, как его малюют - может быть, и не погибнем, но делать будем долго.

Есть готовые примеры. Например, [книга Reza Rad](http://radacad.com/online-book-power-bi-from-rookie-to-rockstar). Сначала он сделал какую-то структуру и постепенно дополнял ее постами и ссылками на свои статьи. Сейчас, насколько могу судить, он убрал пустые ссылки и оставил только заполненные, но ничего не мешает добавлять потом.

Итого, если мы замахиваемся на "справочник Power BI", то можно последовать его примеру. Но, наверное, стоит определиться, что с глубиной проработки. Насколько глубоко копаем? По PQ у меня была начальная идея проработать глубоко использование языка и тонкости/хитрости функций. Сейчас начинаю сомневаться, оно надо ли...

**Comment by hohlick** [comment]
Вот содержание M is for Data Monkey, например

|Глава|Название|Перевод|
|---|---|---|
|Chapter 0|Introduction: A New Revolution||
|Chapter 1|Importing Basic Data|Базовый импорт данных|
|Chapter 2|Overriding Power Query Defaults|Изменение настроек по умолчанию|
|Chapter 3|Basic Append Operations|Базовые операции консолидации|
|Chapter 4|Importing All Files in a Folder|Импорт всех файлов из папки|
|Chapter 5|Aggregating Excel Worksheets|Агрегирование листов Excel|
|Chapter 6|Unpivoting Data|Отмена сведения данных|
|Chapter 7|Importing Nondelimited Text Files|Импорт текстовых файлов без разделителя|
|Chapter 8|Importing from Databases|Импорт из баз данных|
|Chapter 9|Merging Tables and Queries|Слияние таблиц и запросов|
|Chapter 10|Query Loading Destinations|Выгрузка запросов|
|Chapter 11|Defining Data Types|Определение типов данных|
|Chapter 12|Importing Web Data|Импорт данных из Интернет|
|Chapter 13|Loading Data from Exchange|Загрузка данных из Exchange|
|Chapter 14|Grouping and Summarizing|Группировка и агрегирование|
|Chapter 15|Transposing and Unpivoting Complex Data|Транспонирование и разворачивание сложных данных|
|Chapter 16|Automating Refreshing|Автоматическое обновление|
|Chapter 17|Power Query Formulas|Формулы Power Query|
|Chapter 18|Conditional Logic in Power Query|Логические конструкции в Power Query|
|Chapter 19|Power Query Objects|Объекты Power Query|
|Chapter 20|Understanding the M Language|Язык M|
|Chapter 21|Creating Custom Functions|Создание пользовательских функций|
|Chapter 22|Advanced Conditional Logic|Продвинутые логические конструкции|
|Chapter 23|Dynamic Parameter Tables|Динамические таблицы параметров|
|Chapter 24|Dynamic Calendar Tables|Динамические таблицы календаря|
|Chapter 25|Query Organization|Организация запросов|

**Comment by maxim-uvarov** [comment]
Мне кажется, что сейчас гораздо больше нужды в базовых знаниях для входа в отрасль. Т.е. если писать книгу для продвинутых - ей смогут 150 человек воспользоваться, а если писать для новичков, то 3000. 

Мне тоже хочется про продвинутые аспекты рассказывать, но есть понимание, что пользы от хорошего мануала для начинающих будет гораздо больше. 

Мне очень нравится идея просто составлять книгу из блоговых статей. Ну и редактировать их по мере надобности.

**Comment by hohlick** [comment]
Ну, в крайнем случае сможем нарастить. Можно использовать маркировку уровня статьи, типа: noob, intermediate, advanced. Наполнять имеющимся материалом, затем заполнять пробелы по уровням.

**Comment by hohlick** [comment]
Для справки: "Pro Power BI Desktop" (Adam Aspin)

Table of contents

1.	Introduction to Power BI Desktop
2.	Discovering and Loading Data with Power BI Desktop
3.	Transforming Datasets
4.	Data Cleansing
5.	Data Mashup
6.	Creating a Data Model
7.	Extending the Data Model with Calculated Columns
8.	Adding Measures to the Data Model
9.	Analyzing Data over Time
10.	Text-Based Visualizations
11.	Charts in Power BI Desktop
12.	Other Types of Visuals
13.	Filtering Data
14.	Using Slicers
15.	Enhancing Dashboards
16.	PowerBI.com

**Comment by maxim-uvarov** [comment]
я за! ты мой текущий мануал посмотрел? Можно отуда чего взять?

Как по мне так там слабо все. Но вдруг ты чего увидишь. Зато он в маркдауне, хоть сразу берись и вставляй в гитбук

**Comment by maxim-uvarov** [comment]
Макс, мне сейчас в голову пришла идея, что круто будет книгу писать на материалах нашего сообщества - на вопросах оттуда. Тем более что многие уже отвечены. 

Если так двигаться, то все становится гораздо проще - есть потребность, ее уже не надо выдумывать, мы просто описываем ответы на нее.

**Comment by hohlick** [comment]
Текущий мануал смотрел, но там же пдф? Или у тебя исходник в маркдауне?

Про материалы из группы это здорово придумал. Только их классифицировать надо как-то. У тебя были наработки в этом направлении?

**Comment by maxim-uvarov** [comment]
Исходник в маркдауне!

Классифицировать - руками! Можно за основу взять оглавления от коллег, которые ты выше привел. Ну или самостоятельно расфасовывать:)

**Comment by hohlick** [comment]
Тогда предлагаю так. Историю с подробным учебником по PQ оставляю на потом/в фоновый режим.

Делаем общий учебник (новым проектом, чтобы не заниматься переписыванием того, что уже есть для PQ) под рабочим названием "Основы Power BI".

Делаем структуру верхнего уровня, и начинаем наполнять ее всем, до чего дотянемся.

**Comment by maxim-uvarov** [comment]
я за
